src/js/actions/history.js

```diff
@@ -153,6 +154,9 @@ define(function (require, exports) {
                                 count: count,
                                 selectedIndices: selectedIndices
                             });
+                        })
+                        .then(function () {
+                            return this.transfer(toolActions.resetBorderPolicies);
```

Comment:
If it does need to wait for the previous dispatch though, then this can just be left alone.